subsys/bluetooth/audio/csip_set_member.c

```diff
@@ -74,13 +101,63 @@ static bool is_last_client_to_write(const struct bt_csip_set_member_svc_inst *sv
 	}
 }
 
-static void notify_lock_value(const struct bt_csip_set_member_svc_inst *svc_inst,
+static void csip_gatt_notify_complete_cb(struct bt_conn *conn, void *user_data)
+{
+	/* Notification done, clear bit and reschedule work */
+	atomic_clear(&notify_rdy);
+	k_work_submit(&deferred_nfy_work);
+}
+
+static int csip_gatt_notify_set_lock(struct bt_conn *conn,
+			    const struct bt_gatt_attr *attr,
+			    const void *data,
+			    uint16_t len)
+{
+	int err;
+	struct bt_gatt_notify_params params;
+
+	memset(&params, 0, sizeof(params));
+	params.uuid = BT_UUID_CSIS_SET_LOCK;
+	params.attr = attr;
+	params.data = data;
+	params.len  = len;
+	params.func = csip_gatt_notify_complete_cb;
+
+	/* Mark notification in progress */
+	atomic_set(&notify_rdy, 1);
+
+	err = bt_gatt_notify_cb(conn, &params);
+	if (err != 0) {
+		atomic_clear(&notify_rdy);
+
+		if (err == -ENOTCONN) {
+			return err;
+		}
```

Comment:
Should this have been
```suggestion
		if (err != -ENOTCONN) {
			return err;
		}
```

Reply:
Yes :)